ResearchKit/Common/ORKPasscodeStepViewController.m

```diff
@@ -647,4 +672,26 @@ - (BOOL)textFieldShouldEndEditing:(UITextField *)textField {
     return _shouldResignFirstResponder;
 }
 
+- (void)forgotPasscodeTapped
+{
+    if ([self.passcodeDelegate respondsToSelector:@selector(passcodeViewControllerForgotPasscode:)])
+    {
+        [self.passcodeDelegate passcodeViewControllerForgotPasscode:self];
+    }
+}
+
+#pragma mark - Keyboard Notifications
+
+- (void)keyboardWillShow:(NSNotification *)notifcation
+{
+    CGFloat keyboardHeight = [notifcation.userInfo[UIKeyboardFrameEndUserInfoKey] CGRectValue].size.height;
+    
+    double animationDuration = [notifcation.userInfo[UIKeyboardAnimationDurationUserInfoKey] doubleValue];
+    
+    [UIView animateWithDuration:animationDuration animations:^
+    {
+        [_forgotPasscodeButton setFrame:CGRectMake(_forgotPasscodeButton.frame.origin.x, _originalForgotPasscodeY - keyboardHeight, _forgotPasscodeButton.frame.size.width, _forgotPasscodeButton.frame.size.height)];
+    }];
+}
+
```

Comment:
And when the keyboard is dismissed? Or if it goes up and back down? (Not actually sure that it *can*)